docs/samples/custom/torchserve/docs/metrics.md

```diff
@@ -0,0 +1,145 @@
+# Metrics
+
+This adds prometheus and granfana to the cluster with some default metrics.
+
+## Setup
+
+1. Your ~/.kube/config should point to a cluster with [KFServing installed](https://github.com/kubeflow/kfserving/#install-kfserving).
+2. Your cluster's Istio Ingress gateway must be [network accessible](https://istio.io/latest/docs/tasks/traffic-management/ingress/ingress-control/).
+
+## Enable request metrics
+
+```bash
+Kubectl edit cm -n knative-serving config-observability
+```
+
+Add `metrics.request-metrics-backend-destination: prometheus` to data field. You can find detailed information in `data._example` field in the ConfigMap you are editing.
+
+## Add Prometheus and Grafana
+
+Create namespace
+
+```bash
+Kubectl create namespace knative-monitoring
+
+kubectl apply -f https://github.com/knative/serving/releases/download/v0.17.0/monitoring-metrics-prometheus.yaml
+```
+
+## Check pods are up and running
+
+```bash
+kubectl get pods --namespace knative-monitoring
+```
+
+## Forward Prometheus and Grafana ports
+
+```bash
+# Grafana
+kubectl port-forward --namespace knative-monitoring $(kubectl get pods --namespace knative-monitoring \
+--selector=app=grafana --output=jsonpath="{.items..metadata.name}") 3000
+
+# Prometheus
+kubectl port-forward -n knative-monitoring $(kubectl get pods -n knative-monitoring \
+--selector=app=prometheus --output=jsonpath="{.items[0].metadata.name}") 9090
+```
+
+## Access prometheus and grafana
+
+Grafana and Prometheus can be accessed from the below links
+
+Link: [Prometheus](http://localhost:9090/)
+
+Link: [Grafana](http://localhost:3000/)
+
+Logout and login as admin with username: admin and password: admin
+
+## For Torchserve
+
+### config.properties
+
+Enable metrics in config.properties
+
+```json
+metrics_address=http://0.0.0.0:8082
```

Comment:
@jagadeeshi2i Shouldn't this be added in the TorchServe dockerfile as well?

Reply:
Yes 8082 port is exposed in dockerfile added inside torchserve-image folder.